File: Matlab_Source_Code - JT/calcTieredRate.m
function [Cbuy,Base_charge]= calcTieredRate(Base_charge_tier_1,Base_charge_tier_2,Base_charge_tier_3,prices, tierMax, load, daysInMonth)
    Cbuy = zeros(1,8760);
    Base_charge = zeros(1,12);
    hCount = 1;
    for m=1:12
        monthlyLoad = 0;
        for h=1:(24 * daysInMonth(m))
            monthlyLoad= monthlyLoad + load(hCount);
        hCount = hCount + 1;
        end
        totalmonthlyload(m,1)=monthlyLoad;
    end

    hCount = 1;
    for m=1:12
        for h=1:(24 * daysInMonth(m))

            if totalmonthlyload(m,1) < tierMax(1)
                
                Cbuy(hCount) = prices(1);
            elseif totalmonthlyload(m,1) < tierMax(2)
                 
                Cbuy(hCount) = prices(2);
            else
                Cbuy(hCount) = prices(3);
            end
            hCount = hCount+1;
         end
         if totalmonthlyload(m,1) < tierMax(1)
             Base_charge(m) = Base_charge_tier_1;
         elseif totalmonthlyload(m,1) < tierMax(2)
             Base_charge(m) = Base_charge_tier_2;
         else 
             Base_charge(m) = Base_charge_tier_3;
         end 
    end
end

File: Matlab_Source_Code - JT/Python - JT/calcTieredRate.py
from getLoad import getLoad
import numpy as np

def calcTieredRate(base_charge_tier_1, base_charge_tier_2, base_charge_tier_3, prices, tierMax, load, daysInMonth):
    Cbuy = np.zeros(8760)
    base_charge = np.zeros(12)
    totalMonthlyLoad = np.zeros(12)
    hCount = 0

    for m in range(12):
        monthlyLoad = 0
        for h in range(24 * daysInMonth[m]):
            monthlyLoad += load[hCount]
            hCount += 1
        totalMonthlyLoad[m] = monthlyLoad

    hCount = 0
    for m in range(12):
        for h in range(24 * daysInMonth[m]):
            if totalMonthlyLoad[m] < tierMax[0]:
                Cbuy[hCount] = prices[0]
            elif totalMonthlyLoad[m] < tierMax[1]:
                Cbuy[hCount] = prices[1]
            else:
                Cbuy[hCount] = prices[2]
            hCount += 1

        if totalMonthlyLoad[m] < tierMax[0]:
            base_charge[m] = base_charge_tier_1
        elif totalMonthlyLoad[m] < tierMax[1]:
            base_charge[m] = base_charge_tier_2
        else:
            base_charge[m] = base_charge_tier_3

    return [Cbuy, base_charge]

load = getLoad("Load.csv")

tiered = calcTieredRate(1, 2, 3, [0.1, 0.2, 0.3], [500, 1000, 2000], load, [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31])
np.savetxt("Cbuy.csv", tiered[0])
np.savetxt("mc.csv", tiered[1])
